src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -0,0 +1,143 @@
+import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
+import io.github.thebusybiscuit.slimefun4.api.items.settings.IntRangeSetting;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineRecipe;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+/**
+* Represents Book Binder, a machine that binds multiple enchantments books into one.
+*
+* @author ProfElements
+*/
+public class BookBinder extends AContainer {
+
+    private final ItemSetting<Boolean> bypassVanillaMaxLevel = new ItemSetting<>("bypass-vanilla-max-level", false);
+    private final ItemSetting<Boolean> hasCustomMaxLevel = new ItemSetting<>("has-custom-max-level", false);
+    private final ItemSetting<Integer> customMaxLevel = new IntRangeSetting("custom-max-level", 0, 15, Integer.MAX_VALUE);
+
+    @ParametersAreNonnullByDefault
+    public BookBinder(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+        
+        addItemSetting(bypassVanillaMaxLevel, hasCustomMaxLevel, customMaxLevel); 
+    }
+
+    @Override
+    protected MachineRecipe findNextRecipe(BlockMenu menu) {
+        for (int slot : getInputSlots()) {
+            ItemStack target = menu.getItemInSlot(slot == getInputSlots()[0] ? getInputSlots()[1] : getInputSlots()[0]);
+
+            ItemStack item = menu.getItemInSlot(slot);
+            if  (isCompatible(item) && isCompatible(target)) {
+                
+                EnchantmentStorageMeta itemMeta = (EnchantmentStorageMeta) item.getItemMeta();
+                EnchantmentStorageMeta targetMeta = (EnchantmentStorageMeta) target.getItemMeta();
+
+                Map<Enchantment, Integer> storedItemEnchantments = itemMeta.getStoredEnchants();
+                Map<Enchantment, Integer> storedTargetEnchantments = targetMeta.getStoredEnchants();
+                Map<Enchantment, Integer> enchantments = combineEnchantments(storedItemEnchantments, storedTargetEnchantments);
+                
+                if (enchantments.size() > 0) {
+                    ItemStack book = new ItemStack(Material.ENCHANTED_BOOK);
+                    book.setAmount(1);
+
+                    EnchantmentStorageMeta enchantMeta = (EnchantmentStorageMeta) book.getItemMeta();
+                    for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
+                        if (bypassVanillaMaxLevel.getValue()) {
+                            enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+                        } else {
+                            enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), false);
+                        }
```

Comment:
This can be simplified
```suggestion
                        enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), bypassVanillaMaxLevel.getValue());
```